indy_node/server/domain_req_handler.py

```diff
@@ -22,9 +23,172 @@
 logger = getlogger()
 
 
+class RevocationStrategy:
+
+    def __init__(self, state):
+        self.state = state
+        self.author_did = None
+        self.revoc_reg_def_id = None
+        self.req_id = None
+
+    def set_parameters_from_txn(self, author_did, revoc_reg_def_id, req_id):
+        self.author_did = author_did
+        self.revoc_reg_def_id = revoc_reg_def_id
+        self.req_id = req_id
+
+    def set_strategy(self, strategy):
```

Comment:
Why do we need this? We use inheritance for this, don't we? So, it's better to use abstract method to perform strategy-specific validation (do not forget to use @abstractmethod for an abstract class)